### DISCUSSION TITLE
有关弹幕获取的疑问

### DISCUSSION THREAD

**Discussion by Arxhexa** [Nemo2011/bilibili-api#64] [Q&A]
我想问一下对于番剧弹幕的获取，有任何参数可以限制弹幕的总数量吗？

一部比较热门的番剧可能一集的弹幕会有几千行，写入文件也需要大量时间，能否有方法只写入一部分弹幕？

如果只是达成以上目的其实可以随机丢弃一些弹幕，但我想知道b站弹幕的优选（或者原来叫云屏蔽）是单独有另外的api吗？还是每个弹幕都有自己的权重根据需要调整？

**Comment by Nemo2011** [comment]
弹幕优选貌似是在用户弹幕设置里面设置的

**Comment by Arxhexa** [comment]
暂时找到一个方法是从xml源获取，这个源有上限限制

上面说几千行弹幕都说少了，看了下紫罗兰protobuf源能有几万行......

**Comment by Nemo2011** [comment]
是的，xml 源可以设置请求数据大小

**Reply by Nemo2011** [reply]
请问一下，需要写入的文件格式是什么

**Reply by Arxhexa** [reply]
只是调用`make_ass_file_danmakus_protobuf`这个函数，不好意思可能是再之后的xml转ass慢我没再仔细设置时间统计，但如果弹幕确实太多这个函数的总运行时间有时候会超过20秒，而且如果我只是看个番剧几万行的弹幕也确实没有必要

另外能否请教一下xml源如何设置请求数据大小？

**Reply by Nemo2011** [reply]
当然就是设置返回数据的范围啊！最好去除受损数据。
